src/openforms/forms/api/viewsets.py

```diff
@@ -388,6 +391,47 @@ def export(self, request, *args, **kwargs):
         response["Content-Length"] = len(response.content)
         return response
 
+    @extend_schema(
+        summary=_("Set default language for form"),
+        tags=["forms"],
+        request=None,
+        responses={status.HTTP_201_CREATED: FormSerializer},
+        parameters=[
+            UUID_OR_SLUG_PARAMETER,
+            OpenApiParameter(
+                name="Location",
+                type=OpenApiTypes.URI,
+                location=OpenApiParameter.HEADER,
+                description="URL of the created resource",
+                response=[status.HTTP_201_CREATED],
+            ),
+        ],
+    )
+    @transaction.atomic
+    @action(
+        detail=True,
+        methods=["post"],
+        authentication_classes=(),
+        permission_classes=(),
+    )
+    def activate_default_language(self, request, *args, **kwargs):
+        """
+        Set the default language of a form.
+
+        In case translation is not enabled for the Form, the default language is set to
+        settings.LANGUAGE_CODE.
+        """
+        response = Response(status=status.HTTP_200_OK)
+        instance = self.get_object()
+
+        if not instance.translation_enabled:
+            translation.activate(settings.LANGUAGE_CODE)
+            response = set_language_cookie(response, settings.LANGUAGE_CODE)
+
+        response.data = {"active_language": translation.get_language()}
```

Comment:
I'd prefer to see a real serializer used for the response data to keep the implementation/documentation together.

Using https://github.com/open-formulieren/open-forms/blob/master/src/openforms/translations/api/serializers.py:

```py
class CurrentLanguageSerializer(serializers.Serializer):
    active_language = LanguageCodeField()


def activate_default_language(self, request, *args, **kwargs):
    form = self.get_object()
    response = Response(status=status.HTTP_200_OK)
    current_language = translation.get_language()
    if not form.translation_enabled:
        translation.activate(settings.LANGUAGE_CODE)
        current_language = translation.get_language()
        set_language_cookie(response, current_language)

    response.data = CurrentLanguageSerializer(instance={"active_language": current_language}).data
    return response
```